### PR TITLE
Revert "Revert "Load the external javascript and CSS manually when the form is loaded…""

### PR DESCRIPTION
**Pull request by joomlahenk**
Reverts joomlahenk/fabrik#411
Try once again

### CODE REVIEW

File: plugins/fabrik_element/jdate/jdate.php [hunk 1]
```diff
@@ -2666,6 +2666,21 @@ public function formJavascriptClass(&$srcs, $script = '', &$shim = array())
 			$shim[$key] = $s;
 		}
 
+		/* If the jdate element is loaded on an ajax page, we need to manually include the j! calendar  & javascript files */
+		if (FabrikHelperHTML::inAjaxLoadedPage()) {
+			$min = FabrikHelperHTML::isDebug() ? '.min' : '';
+			$jsloc = 'media/system/js/fields/';
```
Comment (trob) [inline plugins/fabrik_element/jdate/jdate.php:2672]: $jsloc = Juri::root() .'media/system/js/fields/';
$cssloc = Juri::root() .'media/system/css/fields/';

File: plugins/fabrik_element/yesno/yesno.php [hunk 2]
```diff
@@ -340,6 +340,27 @@ protected function buttonGroup()
...
+	 * @param   string $script Script to load once class has loaded
+	 * @param   array  &$shim  Dependant class names to load before loading the class - put in requirejs.config shim
+	 *
+	 * @return void
+	 */
+	public function formJavascriptClass(&$srcs, $script = '', &$shim = array())
+	{
+		if (FabrikHelperHTML::inAjaxLoadedPage()) {
+			$min = FabrikHelperHTML::isDebug() ? '.min' : '';
+			echo "<link href='media/system/css/fields/switcher".$min."min.css' rel='stylesheet'>";
```
Comment (trob) [inline plugins/fabrik_element/yesno/yesno.php:356]: echo "<link href='".Juri::root() ."media/system/css/fields/switcher".$min.".css' rel='stylesheet'>";

**Comment by trob** [comment]
It needs the full path to get the correct files in the backend. Otherwise it's looking for ...administrator/media/...

File: plugins/fabrik_element/jdate/jdate.php [hunk 3]
```diff
@@ -2666,6 +2666,21 @@ public function formJavascriptClass(&$srcs, $script = '', &$shim = array())
...
+			$cssloc = 'media/system/css/fields/';
+			echo "<script src='".$jsloc."calendar-locales/date/gregorian/date-helper".$min.".js' type='text/javascript'></script>";
+			echo "<script src='".$jsloc."calendar".$min.".js' type='text/javascript'></script>";
+			echo "<link href='".$cssloc."calendar".$min.".css' rel='stylesheet'>";
+			/* We also need to add them to the shim so they are loaded before the calendar is initialized */
+			$shim[$key]->deps[] = 'media/system/js/fields/calendar.min.js';
```
Comment (trob) [inline plugins/fabrik_element/jdate/jdate.php:2678]: $shim[$key]->deps[] = $jsloc . 'calendar.min.js';
etc.

**Comment by joomlahenk** [comment]
Is not working on my site

File: plugins/fabrik_element/jdate/jdate.php (hunk 3 above)
Comment (joomlahenk) [reply plugins/fabrik_element/jdate/jdate.php:2678]: Strange, on line 2672: $jsloc = 'media/system/js/fields/'; so $shim[$key]->deps[] = $jsloc . 'calendar.min.js'; should work

**Pull request merged**